algorithmic-strategies/divide-and-conquer.md

```diff
@@ -0,0 +1,292 @@
+
+This process is usually done by keeping track of three indices in the array: _starting index_, _ending index_ and _midway index_ as shown in the image below.
+
+![Merge sort example. The list starts with the values (7, 2, 8, 1, 6, 5, 3, 9) (s=0, m=4, e=8). This is split into to lists with one having (7, 2, 8, 1) (s=0, m=2, e=4), and the other having (6, 5, 3, 9) (s=4, m=6, e=8). Each list of four values is split into two lists of 2 value, resulting in 4 total lists. (7, 2) (s=0, m=1, e=2), (8, 1) (s=2, m=3, e=4), (6, 5) (s=4, m=5, e=6), and (3, 9) (s=6, m=7, e=8). Finally, each list of two is split into two single item lists for a total of 8 single item lists. (7) (s=0, e=1), (2) (s=1, e=2), (8) (s=2, e=3), (1) (s=3, e=4), (6) (s=4, e=5), (5) (s=5, e=6), (3) (s=6, e=7), (9) (s=7, e=8). Now each array has only a single value, making it implicitly sorted. The individual subarrays are merged so that they preserve their sorted property by combining them into a temporary array, then copying the sorted values over the merged range until the entire array becomes sorted.](../assets/algorithmic-strategies_divide-and-conquer_merge-sort.png)  
+_Fig. Example run of merge sort showing the index calculations used to split and merge the arrays. Note the convention used of the end index being exclusive._
+
+As we can see in the image above, in the first _divide_ step, the original array of size eight gets divided into two subarrays of size four each. This is done by setting the _starting index_ to _0_, the index of the first element in the array, and the _ending index_ is set to one past the last element in the array, using the convention of the ending index being exclusive. The _midway index_ is then computed using the formula:
+
+<div style="margin:auto;width:fit-content;">
+
+_midway index_ = ⌊(_starting index_ + _ending index_) / 2⌋
+
+</div>
+
+Where ⌊⌋ denotes the mathematical _floor_ operation, or integer truncation.
+
+For the first _divide_ step, the _midway index_ will be _(0+8)/2_ = _4_. In the next _divide_ step, we have two subarrays, one ranging in index from _0_ to _4_ and the other ranging in index from _4_ to _8_. The subarrays are not yet of size one. So, the same action gets repeated to compute the _midway index_. This _divide_ stage continues until the original array of size _n_ is reduced to subarrays of size _1_ each.
+
+A subarray of size one is trivially sorted, since there can be no elements out of place relative to the single value in the array.
+
+The _merge_ stage starts by combining two subarrays at a time. While combining the subarrays containing _7_ and _2_ respectively, the value in each is compared, the smaller value, i.e. _2_, is written to the lower index, and the higher value, i.e. _7_, is written to the higher index. Larger merges would continue comparing the values to be merged element by element, taking the smaller until one subarray is "empty," at which point the remainder of the other array can be copied over.
```

Comment:
Similar to above, puts this in a section and spaces it out a little more. Minimal grammar adjustments.

```suggestion
### Sort and Merge

The _merge_ stage starts by combining two subarrays at a time.

While combining the subarrays containing _7_ and _2_ respectively, the value in each is compared, the smaller value, i.e. _2_, is written to the lower index, and the higher value, i.e. _7_, is written to the higher index.

Larger merges would continue by comparing the values to be merged element by element, taking the smaller until one subarray is "empty." At that point, the remainder of the other array can be copied over!
```

Reply:
added that the subarrays are sorted (since this is the sort and merge section)